pkg/endpoint/endpoint.go

```diff
@@ -2292,6 +2302,12 @@ func (e *Endpoint) Delete(monitor monitorOwner, ipam ipReleaser, manager endpoin
 		}
 	}
 
+	if e.noTrackPort != "" {
+		e.UpdateNoTrackRules(func(ns, podName string) (noTrackPort string, err error) {
+			return "", nil
+		})
+	}
```

Comment:
Did you test this?

`e.Unexpose()` above will disable the eventqueue, so this will be a no-op. From my reading of the code, with this current implementation if you configure the notrackport and then delete an endpoint, it will always report an error in the logs because the event cannot be queued. Then the rules will not be cleaned up.

Reply:
You are absolutely right, this did not work (I overlooked the deletion in my manual test). Changed the logic to be synchronous. Tested in my local cluster:

```
**Before annotation:**
root@gke-nld-default-pool-0a921db7-ppqj:~# iptables-save -c | grep NOTRACK
[0:0] -A CILIUM_OUTPUT_raw -o lxc+ -m mark --mark 0xa00/0xfffffeff -m comment --comment "cilium: NOTRACK for proxy return traffic" -j NOTRACK
[0:0] -A CILIUM_OUTPUT_raw -o cilium_host -m mark --mark 0xa00/0xfffffeff -m comment --comment "cilium: NOTRACK for proxy return traffic" -j NOTRACK
[0:0] -A CILIUM_PRE_raw -m mark --mark 0x200/0xf00 -m comment --comment "cilium: NOTRACK for proxy traffic" -j NOTRACK

**After annotation:**
root@gke-nld-default-pool-0a921db7-ppqj:~# iptables-save -c | grep NOTRACK
[0:0] -A PREROUTING -d 10.80.1.10/32 -p tcp -m tcp --dport 53 -j NOTRACK
[0:0] -A PREROUTING -d 10.80.1.10/32 -p udp -m udp --dport 53 -j NOTRACK
[0:0] -A OUTPUT -d 10.80.1.10/32 -p tcp -m tcp --dport 53 -j NOTRACK
[0:0] -A OUTPUT -s 10.80.1.10/32 -p tcp -m tcp --sport 53 -j NOTRACK
[0:0] -A OUTPUT -d 10.80.1.10/32 -p udp -m udp --dport 53 -j NOTRACK
[0:0] -A OUTPUT -s 10.80.1.10/32 -p udp -m udp --sport 53 -j NOTRACK
[0:0] -A CILIUM_OUTPUT_raw -o lxc+ -m mark --mark 0xa00/0xfffffeff -m comment --comment "cilium: NOTRACK for proxy return traffic" -j NOTRACK
[0:0] -A CILIUM_OUTPUT_raw -o cilium_host -m mark --mark 0xa00/0xfffffeff -m comment --comment "cilium: NOTRACK for proxy return traffic" -j NOTRACK
[0:0] -A CILIUM_PRE_raw -m mark --mark 0x200/0xf00 -m comment --comment "cilium: NOTRACK for proxy traffic" -j NOTRACK

**After deleting the pod directly:**
root@gke-nld-default-pool-0a921db7-ppqj:~# iptables-save -c | grep NOTRACK
[0:0] -A CILIUM_OUTPUT_raw -o lxc+ -m mark --mark 0xa00/0xfffffeff -m comment --comment "cilium: NOTRACK for proxy return traffic" -j NOTRACK
[0:0] -A CILIUM_OUTPUT_raw -o cilium_host -m mark --mark 0xa00/0xfffffeff -m comment --comment "cilium: NOTRACK for proxy return traffic" -j NOTRACK
[0:0] -A CILIUM_PRE_raw -m mark --mark 0x200/0xf00 -m comment --comment "cilium: NOTRACK for proxy traffic" -j NOTRACK
```